File: tests/genColumnIR.test.cpp
#include <fstream>
#include <iostream>
#include <algorithm>

#define CATCH_CONFIG_FAST_COMPILE
#include <catch2/catch.hpp>

#include "../src/lib/loadJSON.hpp"
#include "../src/lib/fileExists.hpp"
#include "../src/lib/searchScrapeTweet.hpp"
#include "../src/lib/bulkTweetStore.hpp"
#include "../src/lib/tweetDate.hpp"
#include "../src/lib/IRGenerator.hpp"

TEST_CASE("should write test data","")
{
    std::ifstream tweetStream("tests/res/bulkStore.json");
    auto json = PanCake::loadJSONFromStream(tweetStream);
    REQUIRE(!json.HasParseError());

    auto tweets = PanCake::parseSearchScrapeTweetsFromJSON(json);

    PanCake::BulkStoreStatus status = PanCake::bulkStoreTweets("tests/rt/columnIR1",tweets);

    REQUIRE(status.success == true);
}

TEST_CASE("should write column IR","")
{
    PanCake::TweetDate date;
    date.year = "2018";
    date.month = "11";
    date.day = "06";

    REQUIRE(PanCake::generateChunkedIR("tests/rt/columnIR1",date,10,"tests/rt/columnIR1") == true);

    REQUIRE(PanCake::fileExists("tests/rt/columnIR1/2018-11-06-sentimentScore.lua"));
    REQUIRE(PanCake::fileExists("tests/rt/columnIR1/2018-11-06-text.lua"));
    REQUIRE(PanCake::fileExists("tests/rt/columnIR1/2018-11-06.chunkCount"));
    REQUIRE(PanCake::fileExists("tests/rt/columnIR1/2018-11-06.totalCount"));


    REQUIRE(PanCake::generateChunkedIR("tests/rt/columnIR1",date,10,"tests/rt/pathThatDoesntExist") == false);
}

File: tests/genColumnIR.test.ts
/// <reference types="jest" />

import * as fs from "fs";

import {bulkStoreTweets} from "../src/include/bulkStoreTweets";
import {genColumnIR} from "../src/include/genColumnIR";
import {compileColumnIR} from "../src/include/compileColumnIR";

jest.setTimeout(10000);

it(`should generate correct IR for columns`,async () => {
    let tweets = fs.readFileSync("tests/res/bulkStore.json").toString();

    await bulkStoreTweets("tests/rt/columnIR2",tweets,"out/");

    await genColumnIR("tests/rt/columnIR2","tests/rt/columnIR2","2018","11","06",10,"out/");

    expect(fs.existsSync("tests/rt/columnIR2/2018-11-06-sentimentScore.lua")).toBe(true);
    expect(fs.existsSync("tests/rt/columnIR2/2018-11-06-text.lua")).toBe(true);
});

it(`should compile IR to bytecode images`,async () => {
    await compileColumnIR("tests/rt/columnIR2/2018-11-06-sentimentScore.lua","tests/rt/columnIR2/2018-11-06-sentimentScore.byc","out/");
    await compileColumnIR("tests/rt/columnIR2/2018-11-06-text.lua","tests/rt/columnIR2/2018-11-06-text.byc","out/");

    expect(fs.existsSync("tests/rt/columnIR2/2018-11-06-sentimentScore.byc")).toBe(true);
    expect(fs.existsSync("tests/rt/columnIR2/2018-11-06-text.byc")).toBe(true);
});